### PR TITLE
Adding support for child gauges

### PR DESCRIPTION
**Pull request by gmbronco**
* BAL rewards in new L2 gauges, AKA child chain gauges are stored as `inflation_rate` variable on the contract. In the gauge syncing workflow we fetch them for all the gauges via multicall and we in the DB as a reward token.

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -435,6 +435,7 @@ model PrismaPoolStakingGauge {
     gaugeAddress        String
     rewards             PrismaPoolStakingGaugeReward[]
     status              PrismaPoolStakingGaugeStatus    @default(ACTIVE)
+    version             Int                             @default(1)
```
Comment (franzns) [inline prisma/schema.prisma:438]: needs to be added to pool.prisma not schema.prisma

schema.prisma is generated by `yarn prisma-merge`

File: modules/token/token.service.ts [hunk 2]
```diff
@@ -121,28 +125,64 @@ export class TokenService {
...
-        //     ...token,
-        //     types: token.types.map((type) => type.type),
-        // }));
+    public async syncCoingeckoIds(): Promise<void> {
+        await this.coingeckoDataService.syncCoingeckoIds();
+    }
 
-        // // Also update any tokens that don't have a coingecko ID by using their contract address
-        // const coingeckoService = new CoingeckoPriceHandlerService(new CoingeckoService());
+    public async getVeBalTotalSupply(): Promise<string> {
```
Comment (danielmkm) [inline modules/token/token.service.ts]: wouldn't really add this here, doesn't make a lot of sense

File: modules/subgraphs/veBal-locks-subgraph/veBal-locks-subgraph.service.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import { GraphQLClient } from 'graphql-request';
+import { networkContext } from '../../network/network-context.service';
+import _ from 'lodash';
+import { VotingEscrowLock_OrderBy, OrderDirection, getSdk } from './generated/veBal-locks-subgraph-types';
+
+export class VeBalLocksSubgraphService {
+    constructor() {}
+
+    // TODO needs proper paging, currently <3000 locks so it works
```
Comment (danielmkm) [inline modules/subgraphs/veBal-locks-subgraph/veBal-locks-subgraph.service.ts]: Would not leave this as a todo, just gonna come back and bite you in the ass later, as you'll forget it

Comment (danielmkm) [reply modules/subgraphs/veBal-locks-subgraph/veBal-locks-subgraph.service.ts]: use `id_gt` as a means to safely paginate

File: modules/token/token.gql [hunk 4]
```diff
@@ -14,6 +14,7 @@ extend type Query {
     tokenGetTokenData(address: String!): GqlTokenData
     tokenGetTokensData(addresses: [String!]!): [GqlTokenData!]!
     tokenGetProtocolTokenPrice: AmountHumanReadable!
+    tokenGetVeBalTotalSupply: AmountHumanReadable!
```
Comment (danielmkm) [inline modules/token/token.gql]: dont put this here

File: modules/token/token.prisma [hunk 5]
```diff
@@ -29,6 +29,8 @@ model PrismaToken {
 
     walletBalances  PrismaUserWalletBalance[]
     stakedBalances  PrismaUserStakedBalance[]
+
+    totalSupply         String?
```
Comment (danielmkm) [inline modules/token/token.prisma]: Same as above, dont think this needs to be added to the token model

File: modules/token/token.resolvers.ts [hunk 6]
```diff
@@ -100,6 +100,9 @@ const resolvers: Resolvers = {
         tokenGetProtocolTokenPrice: async (parent, {}, context) => {
             return tokenService.getProtocolTokenPrice();
         },
+        tokenGetVeBalTotalSupply: async (parent, {}, context) => {
```
Comment (danielmkm) [inline modules/token/token.resolvers.ts]: move this into its own module, not in the tokens module

File: modules/user/lib/user-sync-wallet-balance.service.ts [hunk 7]
```diff
@@ -76,9 +78,100 @@ export class UserSyncWalletBalanceService {
             ],
             true,
         );
+        await this.syncVeBalBalances();
         console.log('initBalancesForPools: finished performing db operations...');
     }
 
+    async syncVeBalBalances() {
```
Comment (danielmkm) [inline modules/user/lib/user-sync-wallet-balance.service.ts]: would also pull this out into its own

**Review by danielmkm — CHANGES_REQUESTED** [review]

**Comment by franzns** [comment]
Thanks, I will create a dedicated veBAL module for all of it.

**Review by danielmkm — APPROVED** [review]

**Pull request merged**